[docs] Adds Workflows documentation (#32840)
expo/expo: docs/pages/workflows/jobs.mdx

---
title: Jobs
description: Learn which pre-packaged and custom jobs you can run with Workflows.
---

Workflows sequence jobs together to automate your development and release processes. You can run two types of jobs: pre-packaged jobs and custom jobs.

> Pre-packaged jobs are provided by Expo and meet common use cases for developers, like building, submitting, and running end to end test. Since they are provided by Expo, they are continuously maintained and updated.

Reviewer: ```suggestion
Expo provides pre-packaged jobs that meet common use cases for developers, such as building, submitting, and running end-to-end tests. Since they are provided by Expo, they are continuously maintained and updated.
```

> Custom jobs are jobs that you write yourself. They are useful for any use case that isn't met by the pre-packaged jobs.

Reviewer: ```suggestion
Custom jobs are jobs that you write yourself. They are useful for any use case not met by the pre-packaged jobs.
```

## Build

This is a pre-packaged job that creates a build from a project:

```yaml .eas/workflows/build.yaml
jobs:
  build:
    type: build
    params:
      platform: android | ios
      profile: string # default: production
```

This job outputs the following variables:

- `platform`: Either `android` or `ios`.
- `build_id`: The ID of the build.

## Submit

This is a pre-packaged job that submits a build to an app store. Given a `build_id`, we'll choose the correct automation to submit the build to the correct store.

```yaml .eas/workflows/submit.yaml
jobs:
  submit:
    type: submit
    params:
      profile: string # default: production
      build_id: string
```

This job outputs the following variables:

- `platform`: Either `android` or `ios`.
- `submission_id`: The ID of the submission.

## Update

This is a pre-packaged job that publishes an update:

```yaml .eas/workflows/update.yaml
jobs:
  update:
    type: update
```

> ## End-to-end test with Maestro

Reviewer: ```suggestion
## End-to-end tests with Maestro
```
